src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/DefaultCSharpCodeActionProvider.cs

```diff
@@ -101,7 +101,7 @@ public DefaultCSharpCodeActionProvider(LanguageServerFeatureOptions languageServ
                 // flag is on, any perf hit here isn't going to affect real users.
                 try
                 {
-                    if (((JToken)codeAction.Data).ToObject<RazorCodeActionResolutionParams>() is not null)
+                    if (((JsonElement)codeAction.Data).Deserialize<RazorCodeActionResolutionParams>() is not null)
```

Comment:
I'm happy to do that in a local function, but I don't really want to put it on a help class anywhere, because I would hate for anyone to think that its a useful function to call. This line is only hit when a feature flag is on, that is only for our information, and only possible to turn on for MS internal users, so the badness of the code doesn't worry me here.